### PR TITLE
Prioritise system installed premake5

### LINKED ISSUE #53: Allow to use system premake
Currently it seems impossible to avoid downloading premake during make instead of using the system installed one. Which also happens to make use of wget which isn't documented as a needed dependency..

### PR DESCRIPTION
**Pull request by marmichalski**
Closes #53. @kattjevfel, can you verify it works for you?

If the system-wide premake5 is not available, the project one will be used & downloaded.

### CODE REVIEW

**Comment by kattjevfel** [comment]
Works as far as it doesn't try to download premake, but it doesn't seem to download it when premake is actually missing:
```
katt@main build/carnage3d (master %) » make build_release
which: no premake5 in (/home/katt/build/scripts/gitrepo:/home/katt/build/scripts:/home/katt/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)
cd third_party/Box2D && \
mkdir -p build && cd build && \
cmake -DCMAKE_BUILD_TYPE=Release -DBOX2D_BUILD_DOCS=Off -DBOX2D_BUILD_UNIT_TESTS=Off -DBOX2D_BUILD_TESTBED=Off .. && \
cmake --build .
...
```

Maybe I am misunderstanding how to test this? It's been a while since I used premake (thank god)

**Comment by marmichalski** [comment]
Maybe you already have it downloaded in the .build dir? :thinking: The output part you have pasted is far before the actual premake check. The interesting output part would be after box2d is actually compiled.

**Comment by kattjevfel** [comment]
Oh, that explains. I really don't wanna mess with building box2d just for this, sorry.

**Comment by marmichalski** [comment]
You do not have to. Makefile builds it for you (it does not install it in your /usr dirs or anywhere else). You can even disable box compilation in the Makefile just to see which premake is being used. :upside_down_face: 

I did some testing on my own, but they're biased, so would be cool if anyone can confirm :sunglasses:

**Comment by kattjevfel** [comment]
I was more referring to wasting resources on building an extra copy of it, but as I've come this far I caved in and got the submodule and built it. It downloaded premake and ran it just like it should (aside from the build craving clang which it won't find any time soon).

tl;dr works as intended
